mirar/references/wfcam/utils.py

```diff
@@ -73,18 +74,25 @@ def get_query_coordinates_from_header(
     return ra_list, dec_list
 
 
-def find_ukirt_surveys(ra: float, dec: float, band: str) -> list[MOCSurvey]:
+def find_wfcam_surveys(
+    ra: float, dec: float, band: str, telescope: str
+) -> list[MOCSurvey]:
     """
     Find which UKIRT survey does the given RA/Dec belong to
     Args:
         :param ra: RA in degrees
         :param dec: Dec in degrees
         :param band: band name
+        :param telescope: telescope name UKIRT or VISTA
 
     Returns:
         :return: list of surveys
     """
-    surveys = get_known_ukirt_surveys()
+    assert telescope.lower() in ["ukirt", "vista"], "Telescope must be UKIRT or VISTA"
```

Comment:
Can you replace this with a more explicit Key Error? I realise assertions are hard to track sometimes. Specifically, please also make sure to mention what value of telescope caused the error.

Reply:
done